privacyidea/lib/resolvers/SQLIdResolver.py

```diff
@@ -356,12 +356,14 @@ def getUserInfo(self, userId):
     
     def _get_userid_filter(self, userId):
         column = getattr(self.TABLE, self.map.get("userid"))
-        if isinstance(column.type, Integer):
+        if isinstance(column.type, String):
+            return column == str(userId)
+        elif isinstance(column.type, Integer):
             # since our user ID is usually a string we need to cast
             return column == int(userId)
-        else:
-            # otherwise we cast the column to string (in case of postgres UUIDs)
-            return cast(column, String).like(userId)
+
+        # otherwise we cast the column to string (in case of postgres UUIDs)
+        return cast(column, String).like(userId)
```

Comment:
Basically this `cast` does the same as the added `if` part, but if i understand this correctly, Oracle expects a size limit of the String when casting (Something like `String(40)`).
Because this is not given, the Oracle db complains about missing parentheses.
We could add something like this here:
```
            try:
                return cast(column, String(column.type.length)).like(userId)
            except AttributeError as _e:
                return cast(column, String(40)).like(userId)
```
which would mitigate the error and we don't have the double String type handling here.

Reply:
So this is only relevant when the id type in the database is a string.
Basically it looks fine and works with postgres as well. Casting with a given string size might get us into different trouble...